### ISSUE TITLE
Go modules not properly tagged

### ISSUE THREAD

**Issue by aurelien-reeves** [cucumber/common#1733]
It seems there may be issues with the release process of go modules.

```console
$ go get github.com/cucumber/common/gherkin/go/v21@v21.0.0
go: github.com/cucumber/common/gherkin/go/v21@v21.0.0: gherkin/go/go.mod has non-.../v21 module path "github.com/cucumber/common/gherkin/go/v20" (and .../v21/go.mod does not exist) at revision gherkin/go/v21.0.0
```

Indeed, the tag [gherkin/go/v21](https://github.com/cucumber/common/blob/gherkin/go%2Fv21.0.0/gherkin/go/go.mod): go.mod for gherkin is versioned as v20.

If we look at the [history of the tag](https://github.com/cucumber/common/commits/gherkin/go/v21.0.0), it has been pushed with when post-releasing messages v17.1.0. It should have been created and pushed later, when releasing gherkin v21.

Actually, the tag [gherkin/v21.0.0](https://github.com/cucumber/common/commits/gherkin/v21.0.0) seems to be good. [gherkin/go/v21.0.0](https://github.com/cucumber/common/commits/gherkin/go/v21.0.0), which is the one used by go to fetch modules, is not.

I review the release scripts, I was not able to spot the issue. Any idea?

**Comment by aurelien-reeves** [comment]
While releasing tag-expressions 4.0.2, I noticed that the git tags for the go package was properly managed so I guess the release process is good.

I'll update the wrong tags manually soon.

**Comment by aurelien-reeves** [comment]
I've fixed gherkin/go/v21.0.0 and json-formatter/go/v19.0.0 tags

Other ones seems to be good.

**Comment by aurelien-reeves** [comment]
We had the same issue today.
The go tag was not set to the good commit, we had to fix it manually.
